hail/python/hailtop/hail_logging.py

```diff
@@ -51,10 +51,31 @@ def log(self, request, response, time):
         now = datetime.datetime.now(tz)
         start_time = now - datetime.timedelta(seconds=time)
         start_time_str = start_time.strftime('[%d/%b/%Y:%H:%M:%S %z]')
+
+        extra = {
+            'remote_address': request.remote,
+            'request_start_time': start_time_str,
+            'request_duration': time,
+            'response_status': response.status,
+            'x_real_ip': request.headers.get("X-Real-IP")
+        }
+
+        batch_identifier = request.get('batch_identifier')
+        if batch_identifier:
+            extra['batch_identifier'] = batch_identifier
+
+        job_identifier = request.get('job_identifier')
+        if job_identifier:
+            extra['job_identifier'] = job_identifier
+
+        batch_operation = request.get('batch_operation')
+        if batch_operation:
+            extra['batch_operation'] = batch_operation
+
+        job_queue_time = request.get('job_queue_time')
+        if job_queue_time:
+            extra['job_queue_time'] = job_queue_time
```

Comment:
what if we add a layer of nesting, say `request['batch_telemetry']['batch_operation']` and then we have `extra={**extra, **request.get('batch_telemetry', {})}`